external/loaders/loaders/batches/_sequences.py

```diff
@@ -145,18 +144,16 @@ def __len__(self) -> int:
         return len(self._args)
 
 
-def shuffle(sequence: Sequence[T], seed: Optional[int] = None) -> Map[T]:
-    """Lazily shuffle a sequence
+def shuffle(sequence: Sequence[T]) -> Map[T]:
+    """Lazily shuffle a sequence. Uses numpy.random for randomness.
 
     Args:
         sequence:  Input sequence to have access indices shuffled
-        seed: Seed for random number generator used for shuffling
     Returns:
         A new shuffled sequence
     """
-    random = RandomState(seed)
     seq_len = len(sequence)
-    shuffled = random.choice(seq_len, size=seq_len, replace=False).tolist()
+    shuffled = np.random.choice(seq_len, size=seq_len, replace=False).tolist()
```

Comment:
It doesn't assume anything, but you're correct - similarly to `np.shuffle` you would have to set the seed at some point prior to calling this for the result to be deterministic.

What you probably don't want to do is set the seed just before each time you call this routine, because while it would give you deterministic shuffles, it would also shuffle identically each time it's called.